modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authorization/authorizer/rdbms/resource/dao/impl/AuthResourceInMemoryDao.java

```diff
@@ -0,0 +1,117 @@
+        inMemoryResourceMap.put(authResource.getResourceType(),
+                                authResource.getResourceName(),
+                                authResource);
+    }
+
+    @Override
+    public void update(AuthResource authResource) throws BrokerAuthServerException {
+        inMemoryResourceMap.put(authResource.getResourceType(),
+                                authResource.getResourceName(),
+                                authResource);
+    }
+
+    @Override
+    public void delete(String resourceType, String resource) throws BrokerAuthServerException {
+        inMemoryResourceMap.remove(resourceType, resource);
+    }
+
+    @Override
+    public AuthResource read(String resourceType, String resource) throws BrokerAuthServerException {
+        return inMemoryResourceMap.get(resourceType, resource);
+    }
+
+    @Override
+    public List<AuthResource> readAll(String resourceType, String ownerId) throws BrokerAuthServerException {
+        Map<String, AuthResource> resourceMap = inMemoryResourceMap.row(resourceType);
+        if (Objects.nonNull(resourceMap)) {
+            Collection<AuthResource> authResources = resourceMap.values();
+            return authResources.stream()
+                                .filter(authResource -> authResource.getOwner()
+                                                                    .equals(ownerId))
+                                .collect(Collectors.toList());
+        }
+        return null;
+    }
+
+    @Override
+    public List<AuthResource> readAll(String resourceType, String action, String ownerId, List<String> userGroups)
+            throws BrokerAuthServerException {
+        Map<String, AuthResource> resourceMap = inMemoryResourceMap.row(resourceType);
+        if (Objects.nonNull(resourceMap)) {
+            Collection<AuthResource> authResources = resourceMap.values();
+            return authResources.stream()
+                                .filter(authResource -> authResource.getOwner().equals(ownerId) ||
+                                        checkActionAndGroups(authResource.getActionsUserGroupsMap(),
+                                                             action,
+                                                             userGroups))
+                                .collect(Collectors.toList());
+        }
+        return null;
+    }
+
+    private boolean checkActionAndGroups(Map<String, Set<String>> actionUserGroupsMap, String action,
+                                         List<String> userGroups) {
+        if (Objects.nonNull(actionUserGroupsMap)) {
+            Set<String> authorizedUserGroups = actionUserGroupsMap.get(action);
+            return Objects.nonNull(authorizedUserGroups) &&
+                    authorizedUserGroups.stream()
+                                        .filter(userGroups::contains)
+                                        .findAny().isPresent();
```

Comment:
Replace this "filter().findAny().isPresent()" chain with "anyMatch()"

Reply:
Fixed in 885868b